enclave_ops/deployment/dkeyserver/App/ecp.cpp

```diff
@@ -101,57 +101,44 @@ bool derive_key(
     sample_sha_state_handle_t sha_context;
     sample_sha256_hash_t key_material;
     
-    memset(&hash_buffer, 0, sizeof(hash_buffer_t));
+    sample_ret = sample_sha256_init(&sha_context);
+    if (sample_ret != SAMPLE_SUCCESS)
+        return false;
 
+    memset(&hash_buffer, 0, sizeof(hash_buffer_t));
```

Comment:
memset_s?

Reply:
not needed. hash_buffer is used @line 110.